internal/internal_workflow_testsuite.go

```diff
@@ -875,16 +875,13 @@ func (env *testWorkflowEnvironmentImpl) Complete(result *commonpb.Payloads, err
 		return
 	}
 	env.workflowDef.Close()
-	var canceledErr *CanceledError
-	if errors.As(err, &canceledErr) && env.workflowCancelHandler != nil {
-		env.workflowCancelHandler()
-	}
```

Comment:
Can you help me understand this change? Is it just because `As` checks errors at every depth and if so, maybe we should change to just checking the exact error type instead of hierarchy? Can you confirm that there is no other internal way for a cancel to be sent to a test workflow than via `TestWorkflowEnvironment.CancelWorkflow()` (which appears to still invoke this callback)/

Reply:
This change is because canceling a  workflow that has already returned is not valid. I the workflow function has already returned what valid effect can the cancel call have?